### ISSUE TITLE
Training diverged error. 

### ISSUE THREAD

**Issue by BalajiVenkatesan006** [justinkay/aldi#9]
[04/22 12:46:39 d2.utils.events]:  eta: 11:56:32  iter: 152  total_loss: 0.7396  loss_cls_source_strong: 0.01294  loss_box_reg_source_strong: 0.02376  loss_rpn_cls_source_strong: 0.005879  loss_rpn_loc_source_strong: 0.06226  loss_cls_distill: 0  loss_box_reg_distill: 0  loss_rpn_cls_distill: 0  loss_rpn_loc_distill: 0  loss_obj_bce_distill: 0.01228  loss_rpn_l1_distill: 0.1409  loss_cls_ce_distill: 0.01911  loss_roih_l1_distill: 0.002177    time: 2.8434  last_time: 2.0695  data_time: 0.8884  last_data_time: 0.4244   lr: 0.06  max_mem: 2313M
Traceback (most recent call last):
  File "/aldi/tools/train_net.py", line 64, in <module>
    launch(
  File "/s/aldi/libs/detectron2/detectron2/engine/launch.py", line 84, in launch
    main_func(*args)
  File "/aldi/tools/train_net.py", line 59, in main
    return trainer.train()
  File "//aldi/libs/detectron2/detectron2/engine/defaults.py", line 484, in train
    super().train(self.start_iter, self.max_iter)
  File "/aldi/libs/detectron2/detectron2/engine/train_loop.py", line 155, in train
    self.run_step()
  File "/aldi/libs/detectron2/detectron2/engine/defaults.py", line 494, in run_step
    self._trainer.run_step()
  File "/aldi/aldi/dropin.py", line 153, in run_step
    loss_dict = self.run_model(data)
  File "/aldi/aldi/trainer.py", line 129, in run_model
    return run_model_labeled_unlabeled(self, *data)
  File "/aldi/aldi/trainer.py", line 104, in run_model_labeled_unlabeled
    do_training_step(labeled_strong, "source_strong", lambda k: do_strong or (do_align and "_da_" in k), do_align=do_align)
  File "aldi/aldi/trainer.py", line 86, in do_training_step
    loss = model(data[batch_i:batch_i+model_batch_size], **kwargs)
  File "/envs/aldi/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1511, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/HPS/oraclase/work/cloud_projects/miniconda3/envs/aldi/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1520, in _call_impl
    return forward_call(*args, **kwargs)
  File "/HPS/oraclase/work/cloud_projects/BalajiThesis/aldi/aldi/rcnn.py", line 25, in forward
    return super(ALDI, self).forward(batched_inputs, do_align=do_align, labeled=labeled)
  File "/aldi/aldi/align.py", line 56, in forward
    output = super().forward(*args, **kwargs)
  File "/aldi/libs/detectron2/detectron2/modeling/meta_arch/rcnn.py", line 161, in forward
    proposals, proposal_losses = self.proposal_generator(images, features, gt_instances)
  File "//miniconda3/envs/aldi/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1511, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/HPS/oraclase/work/cloud_projects/miniconda3/envs/aldi/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1561, in _call_impl
    result = forward_call(*args, **kwargs)
  File "//aldi/libs/detectron2/detectron2/modeling/proposal_generator/rpn.py", line 477, in forward
    proposals = self.predict_proposals(
  File "//aldi/libs/detectron2/detectron2/modeling/proposal_generator/rpn.py", line 503, in predict_proposals
    return find_top_rpn_proposals(
  File "/aldi/libs/detectron2/detectron2/modeling/proposal_generator/proposal_utils.py", line 110, in find_top_rpn_proposals
    raise FloatingPointError(
FloatingPointError: Predicted boxes or scores contain Inf/NaN. Training has diverged.

**Comment by justinkay** [comment]
Thanks, if you can share the config file at `output/oraclase/oraclase_aldi_best/config.yaml` that would also help (it's a bit hard to read the config settings in the log file)

**Comment by BalajiVenkatesan006** [comment]
[config.txt](https://github.com/justinkay/aldi/files/15065288/config.txt)

**Comment by justinkay** [comment]
Hi @BalajiVenkatesan006, anecdotally I ran into this same issue on a new dataset the other day and was able to get around it by reducing the learning rate like so:
```
SOLVER:
  BASE_LR: 0.01
```

Have you had any luck?

**Comment by BalajiVenkatesan006** [comment]
Yes it worked by reducing the learning rate

**Comment by justinkay** [comment]
Great!

**Comment by helia-mohamadi** [comment]
Hello again,
I have also been encountering this error recently (sometimes in the first phase and sometimes in the second phase of training). To test that, I trained a fixed config twice under the same conditions. One of them successfully trained to the end, but the second time, after four hours, it encountered this error: 
FloatingPointError: Predicted boxes or scores contain Inf/NaN. Training has diverged.

**Comment by helia-mohamadi** [comment]
Hello again,
I have also been encountering this error recently (sometimes in the first phase and sometimes in the second phase of training). To test that, I trained a fixed config twice under the same conditions. One of them successfully trained to the end, but the second time, after four hours, it encountered this error:
FloatingPointError: Predicted boxes or scores contain Inf/NaN. Training has diverged.
Also, I saw you reducing the learning rate, but I was wonder to know that why these different situations occur under the same conditions؟

**Comment by helia-mohamadi** [comment]
Hello again,
I have also been encountering this error recently (sometimes in the first phase and sometimes in the second phase of training). To test that, I trained a fixed config twice under the same conditions. One of them successfully trained to the end, but the second time, after four hours, it encountered this error:
FloatingPointError: Predicted boxes or scores contain Inf/NaN. Training has diverged.
Also, I saw you reducing the learning rate, but I was wonder to know that why these different situations occur under the same conditions?
